src/test/java/webdriverFramework/SSMission.java

```diff
@@ -1,53 +1,211 @@
-        driver.findElement(By.xpath("//*[@id='cmp_1']")).click();
 
-//        Select dropDownPodrubrika = new Select(driver.findElement(By.name("'sid'")));
-//        dropDownPodrubrika.selectByValue("1");
+//        WebElement firstAd = electronicsSearchResultPage.allAdvList.get(firstSelectedAd);
+//        WebElement secondAd = electronicsSearchResultPage.allAdvList.get(secondSelectedAd);
+//        WebElement thirdAd = electronicsSearchResultPage.allAdvList.get(thirdSelectedAd);
 
 
+
+
+
+/*
+        List<WebElement> selectedAdsList = new ArrayList<>();
+        selectedAdsList.add(firstAd);
+        selectedAdsList.add(secondAd);
+        selectedAdsList.add(thirdAd);
+
+
+        String firstAdvText = electronicsSearchResultPage.getAdsText(firstAd);
+        String secondAdvText = electronicsSearchResultPage.getAdsText(secondAd);
+        String thirdAdvText = electronicsSearchResultPage.getAdsText(thirdAd);
+
+        Roma thinks this is not necessary anymore. 02/03/17 16:00.
+       */
+
+/*      Roma thinks this is not needed and can be replaced after, 02.03.2017 18:20
+
+        List<String> textsOfSelectedAds = new ArrayList<>();
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[0]));
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[1]));
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[2]));
+
+*/
+        electronicsSearchResultPage.createSelectedAdTextList(selectedAdNumbers); // replaces 4lines above
+
+
+
+
+
+
+
+//        textsOfSelectedAds.add(firstAd.findElement(By.xpath(electronicsSearchResultPage.PATHTOADSTEXT)).getText());
+//          this is not needed right now too
+
+
+
+        electronicsSearchResultPage.selectAdsCheckbox(firstAd);
+        electronicsSearchResultPage.selectAdsCheckbox(secondAd);
+        electronicsSearchResultPage.selectAdsCheckbox(thirdAd);
+
+        electronicsSearchResultPage.showSelectedAdsButton.click();
+
+        ElectronicsShowSelectedPage electronicsShowSelectedPage = new ElectronicsShowSelectedPage(driver);
+
+        List<String> textsOfSelectedFilteredAds = new ArrayList<>();
+        textsOfSelectedFilteredAds.add(electronicsShowSelectedPage.getSelectedAdvText(electronicsShowSelectedPage.selectedAds.get(0)));
+        textsOfSelectedFilteredAds.add(electronicsShowSelectedPage.getSelectedAdvText(electronicsShowSelectedPage.selectedAds.get(1)));
+        textsOfSelectedFilteredAds.add(electronicsShowSelectedPage.getSelectedAdvText(electronicsShowSelectedPage.selectedAds.get(2)));
+
+
+
+
+
+        Collections.sort(textsOfSelectedFilteredAds);
+        Collections.sort(textsOfSelectedAds);
+
+        for (int i = 0; i < textsOfSelectedAds.size(); i++) {
+           assertTrue(textsOfSelectedAds.get(i).contains(textsOfSelectedFilteredAds.get(i)));
+        }
+
     }
 
 
+    public static int randInt(int max) {
+
+        Random rand = new Random();
```

Comment:
In Java 1.7 or later, the standard way to do this is as follows:
`return ThreadLocalRandom.current().nextInt(min, max + 1);`
 see http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html